src/tigerbeetle.zig

```diff
@@ -99,108 +99,120 @@ pub const TransferFlags = packed struct {
 /// When errors do not have an obvious/natural precedence (e.g. "*_must_be_zero"),
 /// the ordering matches struct field order.
 pub const CreateAccountResult = enum(u32) {
-    ok,
-    linked_event_failed,
-    linked_event_chain_open,
-    timestamp_must_be_zero,
-
-    reserved_flag,
-    reserved_field,
-
-    id_must_not_be_zero,
-    id_must_not_be_int_max,
-
-    flags_are_mutually_exclusive,
-
-    ledger_must_not_be_zero,
-    code_must_not_be_zero,
-    debits_pending_must_be_zero,
-    debits_posted_must_be_zero,
-    credits_pending_must_be_zero,
-    credits_posted_must_be_zero,
-
-    exists_with_different_flags,
-    exists_with_different_user_data,
-    exists_with_different_ledger,
-    exists_with_different_code,
-    exists,
+    ok = 0,
+    linked_event_failed = 1,
+    linked_event_chain_open = 2,
+    timestamp_must_be_zero = 3,
+
+    reserved_flag = 4,
+    reserved_field = 5,
+
+    id_must_not_be_zero = 6,
+    id_must_not_be_int_max = 7,
+
+    flags_are_mutually_exclusive = 8,
+
+    ledger_must_not_be_zero = 9,
+    code_must_not_be_zero = 10,
+    debits_pending_must_be_zero = 11,
+    debits_posted_must_be_zero = 12,
+    credits_pending_must_be_zero = 13,
+    credits_posted_must_be_zero = 14,
+
+    exists_with_different_flags = 15,
+    exists_with_different_user_data = 16,
+    exists_with_different_ledger = 17,
+    exists_with_different_code = 18,
+    exists = 19,
+
+    comptime {
+        for (std.enums.values(CreateAccountResult)) |result| {
+            assert(@enumToInt(result) < std.enums.values(CreateAccountResult).len);
+        }
```

Comment:
maybe

```
        for (std.enums.values(CreateAccountResult)) |result, index| {
            assert(@enumToInt(result) == index);
        }
```

?

I think the current assertion only allows permutations, not gaps, but I don't think we would ever _want_ a permutation here anyway?